lib/datadog/appsec/waf.rb

```diff
@@ -609,7 +624,8 @@ def run(input, timeout = LibDDWAF::DDWAF_RUN_TIMEOUT)
           input_obj = Datadog::AppSec::WAF.ruby_to_object(input,
                                                           max_container_size: max_container_size,
                                                           max_container_depth: max_container_depth,
-                                                          max_string_length: max_string_length)
+                                                          max_string_length: max_string_length,
+                                                          coerce: false)
```

Comment:
The change to not coercing the waf input is part of the upgrade to 1.14.0

> In order to ensure that the extracted schema is accurate, tracers must provide the original scalar types to the WAF, rather than just strings, during the rule evaluation stage (ddwaf_run). The rules have been checked to ensure that this will not cause any issues.
This applies to all addresses EXCEPT server.response.status, which needs to be provided as a string. Rules are currently expecting the response code as a string and, since the WAF doesn’t perform type conversions during rule evaluation, providing it as an integer would prevent relevant rules from being evaluated.